crates/ink/ir/src/ir/item_impl/callable.rs

```diff
@@ -324,13 +332,34 @@ where
     if let Some(selector) = callable.user_provided_selector() {
         return *selector
     }
+    let preimage = compose_selector_preimage(item_impl, callable);
+    ir::Selector::compute(&preimage)
+}
+
+/// Returns the composed selector of the ink! callable.
+fn compose_selector_rlp<C>(item_impl: &ir::ItemImpl, callable: &C) -> ir::Selector
+where
+    C: Callable,
+{
+    // todo: handle user provided RLP selector...
+    if let Some(selector) = callable.user_provided_selector() {
```

Comment:
a user provided selector does not rely on encoding, right (equivalent for both scale and RLP)?

If this is the case, this commit should have resolved this TODO: https://github.com/use-ink/ink/pull/2375/commits/5d2f64d40b90b72ada1d8673e615272e4395fbd1